File: core4/leetcode/Arrays/P485/main.cpp
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        
        int streak=0;
        int max_streak=0;
        
        for(int i=0; i<nums.size();i++){
            if (nums[i]==1){
                streak++;
                if (streak > max_streak){
                    max_streak = streak;
                }
            }else{
                streak = 0;
            }
        }
        
        return max_streak;
        
    }
};


int main() {
    vector<int> nums = {1,1,0,1,1,1};

    Solution sol;
    
    cout << "Maximum Consecutive Ones : " << sol.findMaxConsecutiveOnes(nums) << endl;

    return 0;
}

File: core4/leetcode/Arrays/P485/main.py
class Solution:
    def findMaxConsecutiveOnes(self, nums):
        if not nums:
            return 0
        
        max_count = 0
        count = 0
        
        for num in nums:
            if num == 1:
                count += 1
                max_count = max(max_count, count)
            else:
                count = 0
                
        return max_count
        
        
    def findMaxConsecutiveOnes(nums):
        max_count = count = 0
        for num in nums:
            count = count + 1 if num == 1 else 0
            max_count = max(max_count, count)
        return max_count


nums = [1,1,0,1,1,1]
sol = Solution()
print("Maximum Consecutive Ones:", sol.findMaxConsecutiveOnes(nums))
